src/components/editor/property/InputLang.jsx

```diff
@@ -29,66 +24,19 @@ class InputLang extends Component {
     this.props.handleLangChange(payload)
   }
 
-  createOptions = json => json.reduce((result, item) => {
-    // Object.getOwnPropertyDescriptor is necessary to handle the @
-    const id = Object.getOwnPropertyDescriptor(item, '@id').value.replace('http://id.loc.gov/vocabulary/iso639-1/', '')
-    const labelArrayDescr = Object.getOwnPropertyDescriptor(item, 'http://www.loc.gov/mads/rdf/v1#authoritativeLabel')
-
-    // Some of the LOC items do not have labels so ignore them.
-    if (!labelArrayDescr) return result
-    const labelArray = labelArrayDescr.value
-
-    let label = null
-
-    // Looking for English label
-    labelArray.forEach((langItem) => {
-      if (langItem['@language'] === 'en') {
-        label = langItem['@value']
-      }
-    })
-    // But not every language has an English label.
-    if (!label) return result
-
-    result.push({ id, label })
-    return result
-  }, [])
-
   render() {
-    const typeaheadProps = {
-      id: 'langComponent',
-      useCache: true,
-      selectHintOnEnter: true,
-      isLoading: this.state.isLoading,
-      options: this.state.options,
-      selected: this.state.selected,
-      emptyLabel: 'retrieving list of languages...',
-      autoFocus: true,
-    }
-
     return (
       <div>
         <label htmlFor="langComponent">Select language for {this.props.textValue}
           <Typeahead
-            onFocus={() => {
-              if (this.state.isLoading) {
-                return
-              }
-              this.setState({ isLoading: true })
-              fetch('https://id.loc.gov/vocabulary/iso639-1.json')
-                .then(resp => resp.json())
-                .then((json) => {
-                  this.setState({
-                    isLoading: false,
-                    options: this.createOptions(json),
-                  })
-                })
-                .catch(() => false)
-            }}
-            onBlur={() => {
-              this.setState({ isLoading: false })
-            }}
+            onFocus={() => this.props.loadLanguages()}
```

Comment:
How about loading in App.componentDidMount()? That way it is (1) preloaded and (2) available for rendering language button when loading a resource.

Reply:
That's fine as a subsequent change, but this is how it was done previously.